### ISSUE TITLE
Direct LineIn Support

### ISSUE THREAD

**Issue by arminolat** [mdias/rs_asio#327]
Hi there!

I recently started playing RockSmith 2014 again and found out about RS ASIO.
I usually use ASIO4ALL to connect my guitar to my PC using nothing but a 6.3 -> 3.5mm adapter and the results have been amazing in software such as GuitarRig and Amplitube. I'm able to get a clean tone with really low latency (6ms).
I had hopes I might be able to use the same setup with RS ASIO. 
After trying it out, it works, but I have really noticeable lag. Changing Buffer size does nothing, but if I lower Latency Buffer to anything lower than 4, the sounds starts crackling and stuttering.

Is using Line In input only, without interfaces, supported at all? If so, is there anything I might be able to do in order to fix the latency?

Thanks so much!

**Comment by mdias** [comment]
RS ASIO would not be the ideal choice for direct Line In unless your audio card/interface has a native ASIO driver, which judging by what you said does not.

However, if you can use Line-In in other software, it seems you already solved the pre-amp need of the guitar, and so you can use [Rocksmith's Direct Connect mode](https://www.youtube.com/watch?v=H6nAB5ogfeU).

**Comment by arminolat** [comment]
Hi! Thanks for replying!

I tried using direct connect, but that gives me horrible sound, screeching, crackling noise all the time. I can barely hear the guitar sound though all that noise.
In my other programs, both WASAPI and ASIO provide clear sound, but ASIO has way lower latency.
In Rocksmith, only RS ASIO has clear tone, but very noticeable latency (I'd say around 0.5 seconds).

I posted on reddit asking about direct mode and the noise, but got nozhing useful. I would use it if I could get clean tone.
